File: questapp-frontend/src/service/PostService.js
import axios from "axios";

const base_url = "http://localhost:8080/posts";

class PostService {
  async getAllPost() {
    return await axios.get(base_url);
  }
  async createOnePost(post) {
    return await axios.post(base_url, post);
  }
}
export default new PostService();

File: questapp-backend/src/main/java/com/example/questappbackend/services/abs/PostService.java
package com.example.questappbackend.services.abs;

import com.example.questappbackend.dto.PostDtoForCreate;
import com.example.questappbackend.dto.PostDtoForUpdate;
import com.example.questappbackend.dto.responses.PostDtoResponse;
import com.example.questappbackend.entities.Post;

import java.util.List;
import java.util.Optional;

public interface PostService {

    List<PostDtoResponse> getAllPost(Optional<Long> userId);

    Post getOnePost(Long postId);

    Post createOnePost(PostDtoForCreate postRequest);

    Post updateOnePost (Long postId, PostDtoForUpdate updatePost);

    void deleteOnePost(Long postId);
}
